scripts/pylib/twister/twisterlib.py

```diff
@@ -3854,6 +3859,21 @@ def get_testcase(self, identifier):
                     results.append(tc)
         return results
 
+    def verify_platforms_existence(self, platform_names_to_verify, log_info=""):
+        """
+        Verify if platform name (passed by --platform option, or in yaml file
+        as platform_allow or integration_platforms options) is correct. If not -
+        log error.
+        """
+        for platform_name_to_verify in platform_names_to_verify:
```

Comment:
This is too verbose of a variable, please use `for platform in platform_names_to_verify` instead.

I suggest to optimize this, other wise we have 2 for loops here run for each test in worst cases 3 times...

Maybe something like that on top of your change:

```diff
index ce054f9280f..b5309087cc9 100755
--- a/scripts/pylib/twister/twisterlib.py
+++ b/scripts/pylib/twister/twisterlib.py
@@ -3012,6 +3012,8 @@ class TestSuite(DisablePyTestCollectionMixin):
                     logger.error("E: %s: can't load: %s" % (file, e))
                     self.load_errors += 1

+        self.platform_names = [p.name for p in self.platforms]
+
     def get_all_tests(self):
         tests = []
         for _, tc in self.testcases.items():
@@ -3131,7 +3133,7 @@ class TestSuite(DisablePyTestCollectionMixin):
         quarantine_list = []
         for quar_dict in quarantine_yaml:
             if quar_dict['platforms'][0] == "all":
-                plat = [p.name for p in self.platforms]
+                plat = self.platform_names
             else:
                 plat = quar_dict['platforms']
             comment = quar_dict.get('comment', "NA")
@@ -3867,13 +3869,11 @@ class TestSuite(DisablePyTestCollectionMixin):
         as platform_allow or integration_platforms options) is correct. If not -
         log error.
         """
-        for platform_name_to_verify in platform_names_to_verify:
-            for existing_platform in self.platforms:
-                if platform_name_to_verify == existing_platform.name:
-                    break
+        for platform in platform_names_to_verify:
+            if platform in self.platform_names:
+                break
             else:
-                logger.error(f"{log_info} - unrecognized platform - "
-                             f"{platform_name_to_verify}")
+                logger.error(f"{log_info} - unrecognized platform - {platform}")


 class CoverageTool:
```

Reply:
I think that using `platform` instead of `platform_name_to_verify` is oversimplification. `platform` variable has too broad meanings (in Twister context it can mean also `Platform` class object). But it is not a core of this problem, so I just introduced your suggestions.